src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_INDEX = 2;
+    private static final int INCOME_INDEX = 3;
+    private LocalDate startPeriod;
+    private LocalDate endPeriod;
+    private int totalSalary;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[] salaryInfo = new String[names.length];
+        initializeLocalDates(dateFrom, dateTo);
+        for (int i = 0; i < names.length; i++) {
+            for (int y = 0; y < data.length; y++) {
+                String[] info = data[y].split(" ");
+                LocalDate singleDay = LocalDate.parse(info[DATE_INDEX], FORMATTER);
+                if (dateIsValid(singleDay, startPeriod, endPeriod) && nameIsValid(names[i], info)) {
+                    totalSalary += calculateSalary(info);
+                }
+                salaryInfo[i] = names[i] + " - " + totalSalary;
```

Comment:
The line `names[i] + " - " + totalSalary;` is calculated and concatenated in inner loop extra times. We need just increment `salary` in inner loop. And we can concatenate 'name' with salary in the outer loop.